### DISCUSSION TITLE
Support laravel collections that eloquent modifies for full parity (like `only`, `except`, etc)

### DISCUSSION THREAD

**Discussion by myleshyson** [craftcms/cms#15023] [Ideas] [answered]
There are a number of collection methods that Eloquent modifies that Craft should also modify for full support. 

- containsStrict
- diff
- except
- only
- intersect
- unique

Would be nice to able to do something like this.
```php
$data = Entry::find()->section('blog')->collect()->only(['title', 'comments']);
```

**Comment by brandonkelly** [comment]


**Reply by myleshyson** [reply]
I don't know if i fully agree here. For one collections in Craft are arguably mainly used as a collection of Elements. Currently theres a list of basic collection methods that don't work because, unlike how laravel accounts for eloquent models, the Element Collection doesn't account for Element models. 

I actually think it's bad DX to look at the collection documentation, try one of the methods, and realize that it doesn't work because the same considerations weren't taken in the ElementCollection. 

I also think I misspoke, it's not that eloquent is changing behavior, it's just that collections are smart enough to process eloquent models. 

Just seems like a weird stance to choose not to support all collection methods on Element models given thats base of almost everything in Craft. Is it a big deal to have those methods? Not necessarily. But it would be a nice QoL improvement. If anything just so theres not confusion when a Craft developer looks at the collection docs and scratches their head when certain methods don't work.

**Reply by brandonkelly** [reply]
Sorry, I got confused on what `Illuminate\Database\Eloquent\Collection::only()` was doing, and thought it was acting like a multi-value `pluck()`. On second glance that’s not the case. Just going to delete this prior answer…

**Comment by brandonkelly** [comment, marked as answer]
Just added close to full parity with eloquent collections for Craft 4.10 and 5.2 (0fbc77bd4bf30087f5fe401c7cba20dddd6e24b7) 🎉 

See the changelog updates in that commit for full details.

**Reply by myleshyson** [reply]
Whoa awesome that was quick! Thanks for implementing that.

**Reply by brandonkelly** [reply]
Craft 4.10.0-beta.1 and 5.2.0-beta.1 are out with those improvements.
